### PR TITLE
dankomax: added PO methods

### PR DESCRIPTION
**Pull request by dankomax**

### CODE REVIEW

File: src/main/java/com/softserveinc/ita/rozetka/ProductPage.java [hunk 1]
```diff
@@ -1,5 +1,21 @@
...
+
+import static com.codeborne.selenide.Selenide.$x;
+
 public class ProductPage extends BasePage {
-    // implement
+    public String getTitle() {
+        return $x("//h1[@class='product__title']").text();
+    }
+
+    public ShoppingCartModal buy() {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/ProductPage.java:12]: rename

File: src/main/java/com/softserveinc/ita/rozetka/ProductPage.java [hunk 2]
```diff
@@ -1,5 +1,21 @@
...
+        $x("//ul[@class='product-buttons']//button[contains(@class,'buy-button')]").click();
+        return new ShoppingCartModal();
+    }
+
+    public ProductCharacteristicsPage openCharacteristics() {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/ProductPage.java:17]: rename

File: src/main/java/com/softserveinc/ita/rozetka/SearchResultsPage.java [hunk 3]
```diff
@@ -2,8 +2,24 @@
...
+
+    public String getTitle(int product) {
+        return $x(String.format("(//span[@class='goods-tile__title'])[%d]", product)).text();
+    }
+
+    public long getPrice(int product) {
+        return Long.parseLong($x(String.format("(//span[@class='goods-tile__price-value'])[%d]", product)).text());
+    }
+
+    public SearchResultsPage sortDescByPrice() {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/SearchResultsPage.java:20]: rename

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserveinc/ita/rozetka/components/Header.java [hunk 4]
```diff
@@ -1,10 +1,15 @@
...
 import static com.codeborne.selenide.Selenide.$x;
+import static utils.WebElementUtil.isVisible;
 
 public class Header {
     public MobileMenu openMobileMenu() {
         $x("//rz-mobile-user-menu/button").click();
         return new MobileMenu();
     }
+
+    public boolean isShoppingCartEmpty() {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/components/Header.java:12]: ?

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserveinc/ita/rozetka/ProductPage.java (hunk 1 above)
Comment (dankomax) [reply src/main/java/com/softserveinc/ita/rozetka/ProductPage.java]: renamed addToCart

File: src/main/java/com/softserveinc/ita/rozetka/ProductPage.java (hunk 2 above)
Comment (dankomax) [reply src/main/java/com/softserveinc/ita/rozetka/ProductPage.java]: renamed openCharacteristicsPage

File: src/main/java/com/softserveinc/ita/rozetka/SearchResultsPage.java (hunk 3 above)
Comment (dankomax) [reply src/main/java/com/softserveinc/ita/rozetka/SearchResultsPage.java]: renamed sortDescendingByPrice
There is only two sorting options we use in our test cases sortDescendingByPrice and sortAscendingByPrice.

File: src/main/java/com/softserveinc/ita/rozetka/components/Header.java (hunk 4 above)
Comment (dankomax) [reply src/main/java/com/softserveinc/ita/rozetka/components/Header.java]: renamed isCartIconContainsCounter

File: src/main/java/com/softserveinc/ita/rozetka/SearchResultsPage.java [hunk 5]
```diff
@@ -2,8 +2,24 @@
...
+
+    public String getTitle(int product) {
+        return $x(String.format("(//span[@class='goods-tile__title'])[%d]", product)).text();
+    }
+
+    public long getPrice(int product) {
+        return Long.parseLong($x(String.format("(//span[@class='goods-tile__price-value'])[%d]", product)).text());
+    }
+
+    public SearchResultsPage sortDescendingByPrice() {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/SearchResultsPage.java:20]: sortDescending(enum)

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserveinc/ita/rozetka/SearchResultsPage.java (hunk 5 above)
Comment (toleksyn) [reply src/main/java/com/softserveinc/ita/rozetka/SearchResultsPage.java:20]: sortByPrice(ascending)

Comment (dankomax) [reply src/main/java/com/softserveinc/ita/rozetka/SearchResultsPage.java]: created enam ProductSort:
sortBy(PRICE_DESCENDING),
sortBy(POPULARITY), etc.

File: src/main/java/com/softserveinc/ita/rozetka/components/Header.java (hunk 4 above)
Comment (dankomax) [reply src/main/java/com/softserveinc/ita/rozetka/components/Header.java]: renamed isShoppingCartCounterVisible()

**Pull request merged**